### PR TITLE
Gatling EL: random UUID

### PR DESCRIPTION
**Pull request by Amerousful**

### CODE REVIEW

File: gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala [hunk 1]
```diff
@@ -237,6 +237,31 @@ final case class CurrentDateTimePart(format: SimpleDateFormat) extends ElPart[St
   def apply(session: Session): Validation[String] = format.format(new Date()).success
 }
 
+case object randomSecureUUID extends ElPart[UUID] {
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala:240]: Naming rule for classes and non-package objects, capital => `RandomSecureUUID`

File: gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala [hunk 2]
```diff
@@ -237,6 +237,31 @@ final case class CurrentDateTimePart(format: SimpleDateFormat) extends ElPart[St
...
+  def apply(session: Session): Validation[UUID] = UUID.randomUUID().success
+}
+
+case object randomUUID extends ElPart[UUID] {
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala:244]: `randomUUID` => `RandomUUID`

**Review by slandelle — CHANGES_REQUESTED** [review]
Please fix objects naming.
Also, could you please add those new features in the documentation? https://github.com/gatling/gatling/blob/main/src/docs/content/reference/current/core/session/el/index.md#built-in-functions

File: gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala [hunk 3]
```diff
@@ -366,7 +391,11 @@ final class ElCompiler private extends RegexParsers {
 
   private def currentDate: Parser[ElPart[Any]] = "currentDate(" ~> DateFormatRegex <~ ")" ^^ (format => CurrentDateTimePart(new SimpleDateFormat(format)))
 
-  private def nonSessionObject: Parser[ElPart[Any]] = currentTimeMillis | currentDate
+  private def randomSecuredUuid: Parser[ElPart[Any]] = "randomSecuredUuid()" ^^ (_ => RandomSecureUUID)
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala:394]: My bad, it should be `secure`, not `secured`. Sorry for the original typo.

File: src/docs/content/reference/current/core/session/el/index.md [hunk 4]
```diff
@@ -81,6 +81,12 @@ Gatling EL provide the following built-in functions:
   
 // unescape an HTML String (entities decoded)
 "#{foo.htmlUnescape()}"
+
+// generate UUID
```
Comment (slandelle) [inline src/docs/content/reference/current/core/session/el/index.md:85]: generate a random UUID (fast but cryptographically insecure)

File: src/docs/content/reference/current/core/session/el/index.md [hunk 5]
```diff
@@ -81,6 +81,12 @@ Gatling EL provide the following built-in functions:
...
+"#{randomUuid()}"
+
+// generate secure UUID
```
Comment (slandelle) [inline src/docs/content/reference/current/core/session/el/index.md:88]: // generate a random UUID (standard java.util.UUID#randomUUID: cryptographically secure, but slower)

**Review by slandelle — CHANGES_REQUESTED** [review]

File: gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala (hunk 3 above)
Comment (Amerousful) [reply gatling-core/src/main/scala/io/gatling/core/session/el/ElCompiler.scala:394]: Don't worry, my bad because I just copied)

**Review by slandelle — CHANGES_REQUESTED** [review]
One last thing: could you please add some tests in `ElSpec`?

**Comment by Amerousful** [comment]
Okay

**Comment by slandelle** [comment]
> Is it enough just check by type?

Yes. We prove it doesn't crash.

**Review by slandelle — APPROVED** [review]

**Comment by slandelle** [comment]
Cherry-picked, thanks!

**Comment by Amerousful** [comment]
Thank you)

**Pull request closed without merging**